src/binaryutils.cpp

```diff
@@ -158,13 +190,21 @@ bool checkBatchSizeMismatch(const std::shared_ptr<TensorInfo>& tensorInfo,
     return false;
 }
 
-Status validateInput(const std::shared_ptr<TensorInfo>& tensorInfo, const cv::Mat input) {
-    return validateNumberOfChannels(tensorInfo, input);
+Status validateInput(const std::shared_ptr<TensorInfo>& tensorInfo, const cv::Mat input, cv::Mat* firstBatchImage) {
+    // Layout ANY is for pipelines with only custom node as entry.
+    // In this case we need to be sure multiple batch images are aligned to the same resolution.
```

Comment:
Is it required for batching of all shards into one response?
Could that requirement be lifted assuming that based upon DAG configuration (ie parameters of custom node) such DAG should still adhere to such requirement?

Just asking if this is possible improvement for the future.

Reply:
This is outdated comment, will change

Reply:
changed